File: CONCEPTS/persistence.py
import pickle
from dataclasses import dataclass

@dataclass
class StatLine:
    FGM: int
    FGA: int
    FG3M: int
    FG3A: int
    FTM: int
    FTA: int
    
stat1 = StatLine(
    FGM=10, 
    FGA=25, 
    FG3M=4, 
    FG3A=10, 
    FTM=5, 
    FTA=6
)

print('--- StatLine Object ---')
print(stat1)

# Save
with open('stat1.pkl', 'wb') as f:
    pickle.dump(stat1, f)

del stat1  # Remove from memory

# Load stat1
with open('stat1.pkl', 'rb') as f:
   stat1 = pickle.load(f)

print('--- Loaded StatLine Object ---')
print(stat1)

File: CONCEPTS/persistence.R
library(S7)


StatLine <- new_class("StatLine",
  properties = list(
    FGM = class_integer,
    FGA = class_integer,
    FG3M = class_integer,
    FG3A = class_integer,
    FTM = class_integer,
    FTA = class_integer
  )
)

stat1 = StatLine(
  FGM = 10L,
  FGA = 20L,
  FG3M = 3L,
  FG3A = 8L,
  FTM = 5L,
  FTA = 6L
)

print('--- StatLine Instance ---')
print(stat1)

# Save
saveRDS(stat1, "stat1.rds")

# Remove from environment
rm(stat1)

# Load
stat1 <- readRDS("stat1.rds")

print('--- Loaded StatLine Instance ---')
print(stat1)

